bot/database/__init__.py

```diff
@@ -0,0 +1,263 @@
+import typing as t
+from importlib import import_module
+
+import asyncpg
+from loguru import logger
+
+if t.TYPE_CHECKING:
+    from bot.core.bot import Bot
+
+
+class Singleton(type):
+    """
+    This is Singleton Design Pattern.
+
+    It makes sure that classes with this metaclass
+    will only ever have one single instance, when they're
+    initiated for the first time this instance is created,
+    every next initiation will simply result in returning
+    the stored single instace.
+    """
+    _instance = None
+
+    def __call__(cls, *args, **kwargs):
+        """If instance already exists, return it."""
+        if not cls._instance:
+            cls._instance = super(Singleton, cls).__call__(*args, **kwargs)
+
+        return cls._instance
+
+
+class DBTable(metaclass=Singleton):
```

Comment:
What's `metaclass` ?

Reply:
A class which determines how other classes work internally, more on that [here](https://realpython.com/python-metaclasses/)